modules/layers/src/layers/editable-h3-cluster-layer.ts

```diff
@@ -0,0 +1,179 @@
+/* eslint-env browser */
+
+import { H3ClusterLayer } from '@deck.gl/geo-layers';
+import { ViewMode } from '@nebula.gl/edit-modes';
+import { polyfill, geoToH3 } from 'h3-js';
+import EditableGeoJsonLayer from './editable-geojson-layer';
+import EditableLayer from './editable-layer';
+
+const DEFAULT_EDIT_MODE = ViewMode;
+const DEFAULT_H3_RESOLUTION = 9;
+
+const defaultProps = {
+  mode: DEFAULT_EDIT_MODE,
+
+  // EditableGeoJsonLayer
+  geoJsonData: {},
+  ...EditableGeoJsonLayer.defaultProps,
+
+  // h3 layer
+  data: [],
+  selectedIndexes: [],
+  filled: false,
+  stroked: true,
+  lineWidthScale: 1,
+  lineWidthMinPixels: 1,
+  lineWidthMaxPixels: Number.MAX_SAFE_INTEGER,
+  lineWidthUnits: 'pixels',
+  getHexagons: (d) => d.hexIds,
+  resolution: DEFAULT_H3_RESOLUTION,
+};
+
+export default class EditableH3ClusterLayer extends EditableLayer {
+  static layerName = 'EditableH3ClusterLayer';
+  static defaultProps = defaultProps;
+
+  initializeState() {
+    super.initializeState();
+
+    this.setState({
+      tentativeHexagonIDs: [],
+    });
+  }
+
+  // convert array of (lng, lat) coords to cluster of hexes
+  getDerivedHexagonIDs(coords) {
+    return polyfill(coords, this.props.resolution, true);
+  }
+
+  // convert pair of (lng, lat) coords into single hex
+  getDerivedHexagonID(coords) {
+    return geoToH3(coords[1], coords[0], this.props.resolution);
+  }
+
+  renderLayers() {
+    const layers: any = [
+      new EditableGeoJsonLayer(
+        this.getSubLayerProps({
+          id: 'editable-geojson',
+
+          mode: this.props.mode,
+          data: {
```

Comment:
this may cause a performance issue since deck.gl will see data as a new object on every render. May need to extract this as a const.

```
const EMPTY_FEATURE_COLLECTION = {...}

export class EditableH3ClusterLayer {
...
  new EditableGeoJsonLayer({
    ...
    data: EMPTY_FEATURE_COLLECTION
  ...
}